src/Orleans/Messaging/GatewayClientReceiver.cs

```diff
@@ -58,13 +58,14 @@ protected void RunNonBatch()
             }
             catch (Exception ex)
             {
+                buffer.Reset();
```

Comment:
given this usage, shouldn't the Reset() be returning all the buffers to the pool, in case it's not used ever again?